ansible/modules/hashivault/hashivault_db_secret_engine_config.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python
+from ansible.module_utils.hashivault import hashivault_argspec
+from ansible.module_utils.hashivault import hashivault_auth_client
+from ansible.module_utils.hashivault import hashivault_init
+from ansible.module_utils.hashivault import hashiwrapper
+import json, sys
+
+ANSIBLE_METADATA = {'status': ['stableinterface'], 'supported_by': 'community', 'version': '1.1'}
+DOCUMENTATION = '''
+---
+module: hashivault_db_secret_engine_config
+version_added: "3.17.8"
+short_description: Hashicorp Vault database secrets engine config
+description:
+    - Module to configure a database secrets engine 
+options:
+    url:
+        description:
+            - url for vault
+        default: to environment variable VAULT_ADDR
+    ca_cert:
+        description:
+            - "path to a PEM-encoded CA cert file to use to verify the Vault server TLS certificate"
+        default: to environment variable VAULT_CACERT
+    ca_path:
+        description:
+            - "path to a directory of PEM-encoded CA cert files to verify the Vault server TLS certificate : if ca_cert
+             is specified, its value will take precedence"
+        default: to environment variable VAULT_CAPATH
+    client_cert:
+        description:
+            - "path to a PEM-encoded client certificate for TLS authentication to the Vault server"
+        default: to environment variable VAULT_CLIENT_CERT
+    client_key:
+        description:
+            - "path to an unencrypted PEM-encoded private key matching the client certificate"
+        default: to environment variable VAULT_CLIENT_KEY
+    verify:
+        description:
+            - "if set, do not verify presented TLS certificate before communicating with Vault server : setting this
+             variable is not recommended except during testing"
+        default: to environment variable VAULT_SKIP_VERIFY
+    authtype:
+        description:
+            - "authentication type to use: token, userpass, github, ldap, approle"
+        default: token
+    token:
+        description:
+            - token for vault
+        default: to environment variable VAULT_TOKEN
+    username:
+        description:
+            - username to login to vault.
+        default: to environment variable VAULT_USER
+    password:
+        description:
+            - password to login to vault.
+        default: to environment variable VAULT_PASSWORD
+    mount_point:
+        description:
+            - name of the secret engine mount name.
+        default: database
+    state:
+        description:
+            - should configuration be present or absent
+        default: present
+    config_file:
+        description:
+            - optional: location of file containing relevant db configuration info. use either this or the following ansible params in your play
```

Comment:
Can't really hijack `username` and `password`, probably should be `database_user` and `database_password` or some such

Reply:
ah. the userpass auth method...

Reply:
so based on my understanding of the db engine... it may HAVE to hijack it https://www.vaultproject.io/docs/secrets/databases/index.html

username and password appear to be reserved words for the engine to interpolate

Reply:
so what i will do is set the ansible param to `db_username` / `db_password` and after the client is authed, they will overwrite the values in desired state